### PR TITLE
Simplify jdonslurm

### PR DESCRIPTION
**Pull request by cpranav93**
Draft PR for simplifying installation

### CODE REVIEW

File: installJDOnSLURM.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+def test_mamba(conn):
+    """
+    Check if mamba is installed on remote host
+
+    :param conn: ssh connection object
+    :return mamba_exists: Logical on if mamba is installed on remote host
+    """
+    cmd = "which mamba"
+    result = conn.run(cmd, hide=True)
```
Comment (meiertgrootes) [inline installJDOnSLURM.py:65]: running from mac on pristine SPIDER surf account causes fatal exit

`
File "/Users/eslt0101/mambaforge/envs/RS-DAT/lib/python3.9/site-packages/invoke/runners.py", line 509, in _finish
    raise UnexpectedExit(result)
invoke.exceptions.UnexpectedExit: Encountered a bad command exit code!

Command: 'which mamba'

Exit code: 1

Stdout:



Stderr:

/usr/bin/which: no mamba in (/home/stursdat-mgrootes/.local/bin:/home/stursdat-mgrootes/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin)
`
it seems that invoke.run (which is wrapped by connection.run) doesn't  pass exceptions out but raises them. Did you test this? Is this only me?

**Review by meiertgrootes — CHANGES_REQUESTED** [review]
Looks great overall. We can take a separate stab at making file paths more configurable (-> new feature branch for that). I get a fatal error on running however (see specific comment).

File: installJDOnSLURM.py (hunk 1 above)
Comment (cpranav93) [reply installJDOnSLURM.py:65]: This works on windows WSL without raising a fatal exit. Maybe there is a better way to check if mamba is installed. Any thoughts on how we can?

File: runJupyterDaskOnSLURM.py [hunk 2]
```diff
@@ -379,24 +379,40 @@ def forward_port_and_launch_local(conn,forwardconfig):
...
+    # Check and install on remote as needed
+    user_install = input('Are required components already installed on remote host? (Y/n): ')
+    if user_install == 'Y':
+        install = True
     else:
-        _ = ssh_remote_executor(config_inputs,forward_port_and_launch_local, forwardconfig)
-
+        install = installJDOnSLURM.install_JD(config_inputs, platform_name, envfile = 'environment.yaml')
+
+    user_uninstall = input('Do you want to uninstall all components on remote host? (Y/n): ')
```
Comment (fnattino) [inline runJupyterDaskOnSLURM.py:397]: I had an issue here, as I typed "y" (lower-case) and this was not recognised as a yes. Also, the "(Y/n)" prompt suggests "Y" to be the default, but actually there is no default. Something like the following could fix these issues?
```suggestion
    user_uninstall = (input('Do you want to uninstall all components on remote host? (Y/n): ') or "Y").upper()
```
We might then also have to catch the case in which one types in something different from "y" or "n".

File: installJDOnSLURM.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        ssh_remote_executor(config_inputs, create_env, envfile)
+        env_exists = ssh_remote_executor(config_inputs, test_env, envfile)
+        if not env_exists:
+            raise ValueError(f'Error creating environment. Please create manually')
+        
+    #Configure Jupyter
+    jpconfig_exists = ssh_remote_executor(config_inputs, check_jpconfig)
+    if not jpconfig_exists:
+        print ('Configuring Jupyter on remote host...')
+        ssh_remote_executor(config_inputs, jpconfig, envname)
```
Comment (fnattino) [inline installJDOnSLURM.py:197]: Had some issues in filling in the password - it asked for this repeatedly, and finally took in only an empty string. Anybody else had the same issue?

**Review by fnattino — COMMENTED** [review]
Good way to do the initial configuration. I had two hiccups, for which I have added comments below. 

Considering the current size of the scripts and  the shared functions, we could think about refactoring them at some point in something more "modular", and a different command-line interface maybe along the line of:
```shell
jupyter-dask-slurm config # add platform
jupyter-dask-slurm install -p platform # install
jupyter-dask-slurm run -p platform # run
```

**Comment by cpranav93** [comment]
Ou has tested the other PR which includes this. So I am merging this as well.

File: installJDOnSLURM.py (hunk 3 above)
Comment (meiertgrootes) [reply installJDOnSLURM.py:197]: Couldn't replicate the issue running from mac. Pinging @cpranav93  just fyi

**Pull request merged**